cmake/Modules/FindComputeCpp.cmake

```diff
@@ -397,19 +396,44 @@ function(add_sycl_to_target)
     "${multi_value_args}"
     ${ARGN}
   )
-  set(fileCounter 0)
-  # Add custom target to run compute++ and generate the integration header
-  foreach(sourceFile ${SDK_ADD_SYCL_SOURCES})
-    if(NOT IS_ABSOLUTE ${sourceFile})
-      set(sourceFile "${CMAKE_CURRENT_SOURCE_DIR}/${sourceFile}")
+
+  # If the CXX compiler is set to compute++ enable the driver. This removes the
+  # need to explicitly generate the sycl integration header in a separate
+  # compiler pass.
+  get_filename_component(cmakeCxxCompilerFileName "${CMAKE_CXX_COMPILER}" NAME)
+  if("${cmakeCxxCompilerFileName}" STREQUAL "compute++")
+    if(MSVC)
+      message(FATAL_ERROR "The compiler driver is not supported by this system,
+                           revert the CXX compiler to your default host compiler.")
     endif()
-    __build_ir(
-      TARGET     ${SDK_ADD_SYCL_TARGET}
-      SOURCE     ${sourceFile}
-      COUNTER    ${fileCounter}
-    )
-    MATH(EXPR fileCounter "${fileCounter} + 1")
-  endforeach()
+
+    # Set all flags required to compile the target when using the compiler driver
+    list(APPEND computecppDriverCompileFlags ${COMPUTECPP_DEVICE_COMPILER_FLAGS}
+                                             ${COMPUTECPP_USER_FLAGS})
+    get_target_property(includeAfter ${SDK_ADD_SYCL_TARGET} COMPUTECPP_INCLUDE_AFTER)
+    if(includeAfter)
+      list(APPEND computecppDriverCompileFlags -fsycl-ih-last)
+    endif()
+    # Remove flags that are handled by the driver.
+    list(REMOVE_ITEM computecppDriverCompileFlags -emit-llvm -sycl)
```

Comment:
We get these flags from the computecpp_info tool, but it's starting to look like we might as well maintain our own list here, since the computecpp_info ones are occasionally outdated (like -emit-llvm). That way, you'd not have to remove any here.

Reply:
Yeah I'm not a big fan of the flags given by computecpp_info to be honest.
It would break a lot of existing CMake samples if we stop using them by default though :/

Reply:
Nevermind ignore me it would still work just fine. We could just by default add `-O2 -mllvm -inline-threshold=1000 -intelspirmetadata` to `COMPUTECPP_DEVICE_COMPILER_FLAGS` and later either add `-sycl-driver` or `-sycl`. Would this mean we would be able to remove the call to `computecpp_info --dump-is-supported` as well as `--dump-device-compiler-flags`?

Reply:
The is-supported thing is arguably becoming less useful, as it will tell you some platforms aren't supported when they are. And it doesn't make sense when cross-compiling either. And, in fact, I don't know how it determines which platform you're targeting (you might have several, only one of which is supported). It's possible, then, that removing a couple of those calls wouldn't be a bad thing. It makes the followup code so much simpler (`-sycl` can then be added only inside build_ir, for example). I'd like to keep `computecpp_info` as a part of the whole find process, though, users might still want to do whatever with it.

Reply:
I only removed the call to --dump-device-compiler-flags as I think --dump-is-supported is unrelated to this PR after all.

Reply:
It is unrelated, but I still wouldn't mind it being removed :p